dashboard/lib/feature_mode_manager.rb

```diff
@@ -111,6 +111,39 @@ def self.get_mode(gatekeeper, dcdo, hoc_script_names)
     end
   end
 
+  # Returns a set of the names of all of the Gatekeeper features enabled or disabled for
+  # the given mode.
+  def self.get_feature_names_for_mode(mode)
+    settings = MODE_SETTINGS_MAP[mode]
+    Set.new(settings[:gatekeeper_general_settings].keys +
+                settings[:gatekeeper_hoc_tutorial_settings].keys)
+  end
+
+  # Returns true if the given mode allows the given feature for a hoc script, false if
+  # it disallows it, and nil if the setting is unspecified.
+  def self.mode_allows_feature_for_hoc_scripts(mode, feature)
+    return nil unless MODES.include?(mode)
+    MODE_SETTINGS_MAP[mode][:gatekeeper_hoc_tutorial_settings][feature]
+  end
+
+  # Returns true if the given mode allows the given feature by default.
+  def self.mode_allows_feature_by_default(mode, feature)
+    return nil unless MODES.include?(mode)
+    MODE_SETTINGS_MAP[mode][:gatekeeper_general_settings][feature]
+  end
+
+  # Returns true if the feature mode manager allows the `feature` in `mode`,
+  # falling through to the value from the Gatekeeper if the manager doesn't specify a value.
+  def self.allows(gatekeeper, mode, feature, script)
+    setting =
+        FeatureModeManager.mode_allows_feature_for_hoc_scripts(mode, feature) ||
```

Comment:
Style/IndentAssignment: Indent the first line of the right-hand-side of a multi-line assignment.